MikuMikuLibrary/Objects/Processing/Assimp/AssimpImporter.cs

```diff
@@ -212,71 +212,95 @@ private static Mesh CreateMeshFromAiNode(Ai.Node aiNode, Ai.Scene aiScene, Objec
 
                     // Blender has this weird quirk where it duplicates bones.
                     // We can handle it by grouping bones of the same name.
+
                     foreach (var boneGroup in aiMesh.Bones.GroupBy(x => x.Name))
                     {
                         // Likely going to have duplicates! Order by vertex ID so we can detect them.
-	                    var vertexWeights = boneGroup
-		                    .SelectMany(x => x.VertexWeights).OrderBy(x => x.VertexID).ToList();
+                        var vertexWeights = boneGroup
+                            .SelectMany(x => x.VertexWeights).OrderBy(x => x.VertexID).ToList();
 
                         // Skip if empty (apparently those also exist when exporting from Blender)
                         if (vertexWeights.Count == 0)
-	                        continue;
+                            continue;
 
                         var aiBone = boneGroup.First();
 
                         int boneIndex = obj.Skin.Bones.FindIndex(
-	                        x => x.Name == aiBone.Name);
+                            x => x.Name == aiBone.Name);
 
                         if (boneIndex == -1)
                         {
-	                        boneIndex = obj.Skin.Bones.Count;
-
-	                        var aiBoneNode = aiScene.RootNode.FindNode(aiBone.Name);
-
-	                        // This is not right, but I'm not sure how to transform the bind pose matrix
-	                        // while not having duplicate bones.
-	                        Matrix4x4.Invert(GetWorldTransform(aiBoneNode), out var inverseBindPoseMatrix);
-
-	                        obj.Skin.Bones.Add(new BoneInfo
-	                        {
-		                        Name = aiBoneNode.Name,
-		                        InverseBindPoseMatrix = inverseBindPoseMatrix
-	                        });
+                            var aiBoneNode = aiScene.RootNode.FindNode(aiBone.Name);
+
+                            void AddParents(Ai.Node aiBoneNode)
+                            {
+                                Ai.Node parentNode = aiScene.RootNode.FindNode(aiBoneNode.Parent.Name);
+                                if (parentNode != null && !parentNode.HasMeshes && parentNode != aiScene.RootNode && (!aiScene.RootNode.Children.Contains(parentNode)))
+                                {
+                                    if (obj.Skin.Bones.Find(x => x.Name == parentNode.Name) == null)
```

Comment:
You could use .Any here like so:
```suggestion
                                    if (!obj.Skin.Bones.Any(x => x.Name == parentNode.Name))
```

Reply:
Funny story, i was trying to use .Any in [180a78a](https://github.com/blueskythlikesclouds/MikuMikuLibrary/pull/61/commits/180a78acdb929ef18cfa2093afd9130fafa1c234) but i was getting strange results. i guess it was just a byproduct of the attempt at parenting directly when adding not working. Updated.